### PR TITLE
Update and restructure environments

### PR DESCRIPTION
**Pull request by melindafekete**
## About the changes
Update and restructure Environments documentation. The main changes are:
- Environments concepts page:
  - Updated and restructured content and applied writing style changes
  - Added more explanations and use cases
  - Added more links to other related concepts
  - Removed the outdated Loom video from Ivar
  - Removed outdated sections (how to 'start' using environments, and migration)
  - Removed screenshots and used a better-fitting diagram to illustrate strategies and environments
  - Removed the list of events, as these are well documented on the [events page](https://docs.getunleash.io/reference/events)
- Removed the [How to clone environments](https://docs.getunleash.io/how-to/how-to-clone-environments) guide. This content is explained on the concepts page, and I felt there was not enough content to fill a standalone guide for this.

Preview link of the updated [Environments page](https://unleash-docs-git-melinda-environments-docs-unleash-team.vercel.app/reference/environments).

### CODE REVIEW

File: website/docs/reference/environments.md [hunk 1]
```diff
@@ -8,112 +8,63 @@ title: Environments
...
 
-- **/api/admin/features** - _deprecated (scheduled for removal in Unleash v5.0). The [old feature flags admin API](/reference/api/legacy/unleash/admin/features.md) still works, but strategy configuration will be assumed to target the “default” environment._
-- **/api/admin/projects/:projectId/features** - New feature API to be used for feature flags which also adds support for environments. See [the documentation](/reference/api/legacy/unleash/admin/features.md) to learn more.
+Cloning an environment creates a copy of the selected source environment. You can select which projects to clone the environment configuration in. This means that the default visibility of the source environment, as well as all feature flag configurations of the source environment, are cloned in the new environment. For example, if a feature flag had a gradual rollout to 67% in the source environment, then your new environment will have the same feature enabled with the same activation strategy in the new environment. After cloning, the environments are independent and do not stay in sync.
```
Comment (melindafekete) [inline website/docs/reference/environments.md:68]:
```suggestion
Cloning an environment creates a copy of the selected source environment. You can select which projects to clone the environment configuration in. This means that the default visibility of the source environment, as well as any feature flag configurations, are cloned in the new environment. For example, if a feature flag had a gradual rollout to 67% in the source environment, then your new environment will have the same feature enabled with the same activation strategy in the new environment. After cloning, the environments are independent and do not stay in sync.
```

File: website/docs/reference/environments.md [hunk 2]
```diff
@@ -8,112 +8,63 @@ title: Environments
...
+Each environment has a name and one of the following types:
 - Development
 - Test
 - Pre-production
 - Production
 
-The **production** environment type is special: Unleash will show additional confirmation prompts when you change something that could impact users in environments of this type. The built-in "production" environment is a production-type environment.
-
-The other environment types do not currently have any functionality associated with them. This may change in the future.
+Environments of type production show additional confirmation prompts when making changes that may affect users. The default **production** environment is of type production.
```
Comment (FredrikOseberg) [inline website/docs/reference/environments.md:27]: It also determines when a feature goes into the **live** lifecycle step. But not sure if we need to mention that here.

**Review by FredrikOseberg — APPROVED** [review]
These changes look good to me, I've added some optional comments you can choose whether or not to address before merging.

File: website/docs/reference/environments.md (hunk 2 above)
Comment (melindafekete) [reply website/docs/reference/environments.md]: Agreed, thanks! fabfdcc09a4d2f8e46049a4c7377791faab698c2

Comment (FredrikOseberg) [reply website/docs/reference/environments.md]: Small note on the change in terms of feature lifecycle: 

* Live is reached for the feature when it is receiving metrics AND is exposed to users. In other words, some of the metrics have to report exposure.
* For pre-live it's enough to report metrics from any non-production environment

Comment (melindafekete) [reply website/docs/reference/environments.md]: thanks, I'll make that clarification

**Pull request merged**